commit 218cce8e53e378ab5f269636d768ac1c7aa1f70b
Author: Andrea Iacovitti <aiacovitti@libero.it>
Date:   Sat Nov 2 13:16:31 2013 +0100

    Remove the object from the floatingObject list too in RenderObject::removeFromObjectLists,
    otherwise FloatingObject.node become a dangling pointer.
    
    BUG: 302519
    BUG: 215719
    FIXED-IN: 4.11.4

diff --git a/khtml/rendering/render_object.cpp b/khtml/rendering/render_object.cpp
index fddc029..7185469 100644
--- a/khtml/rendering/render_object.cpp
+++ b/khtml/rendering/render_object.cpp
@@ -2380,6 +2380,13 @@ void RenderObject::removeFromObjectLists()
 
         if (outermostBlock)
             outermostBlock->markAllDescendantsWithFloatsForLayout(this);
+
+        RenderObject *p;
+        for (p = parent(); p; p = p->parent()) {
+            if (p->isRenderBlock())
+                static_cast<RenderBlock*>(p)->removeFloatingObject(this);
+        }
+
     }
 
     if (isPositioned()) {
